landoapi/transplants.py

```diff
@@ -312,46 +312,47 @@ def warning_code_freeze(*, repo, **kwargs):
     except KeyError:
         return
 
-    if repo_details.product_details_url:
-        default_message = "Could not retrieve repository's code freeze status."
-
-        try:
-            product_details = requests.get(repo_details.product_details_url).json()
-        except requests.exceptions.RequestException as e:
-            logger.exception(e)
-            return [{"message": default_message}]
-
-        # If the JSON doesn't have these keys, this warning isn't applicable
-        try:
-            freeze_date_str = product_details["NEXT_SOFTFREEZE_DATE"]
-            merge_date_str = product_details["NEXT_MERGE_DATE"]
-        except KeyError:
-            return
-
-        # The code freeze dates generally correspond to PST work days.
-        utc_offset = "-0800"
-        today = datetime.now(tz=timezone.utc)
-        freeze_date = datetime.strptime(
-            f"{freeze_date_str} {utc_offset}",
-            "%Y-%m-%d %z",
-        ).replace(tzinfo=timezone.utc)
-        if today < freeze_date:
-            return
-
-        merge_date = datetime.strptime(
-            f"{merge_date_str} {utc_offset}",
-            "%Y-%m-%d %z",
-        ).replace(tzinfo=timezone.utc)
-
-        if freeze_date <= today <= merge_date:
-            return [
-                {
-                    "message": (
-                        f"Repository is under a soft code freeze "
-                        f"(ends {merge_date_str})."
-                    )
-                }
-            ]
+    if not repo_details.product_details_url:
+        # Repo does not have a product details URL.
+        return
+
+    try:
+        product_details = requests.get(repo_details.product_details_url).json()
+    except requests.exceptions.RequestException as e:
+        logger.exception(e)
+        return [{"message": "Could not retrieve repository's code freeze status."}]
+
+    # If the JSON doesn't have these keys, this warning isn't applicable
+    try:
+        freeze_date_str = product_details["NEXT_SOFTFREEZE_DATE"]
+        merge_date_str = product_details["NEXT_MERGE_DATE"]
+    except KeyError:
+        return
```

Comment:
At this point in the refactor, I think this would be a little cleaner and more explicit, as well as covers the case when the input is null/blank (if that's ever the case):

```suggestion
freeze_date_str = product_details.get("NEXT_SOFTFREEZE_DATE")
merge_date_str = product_details.get("NEXT_MERGE_DATE")

if not freeze_date_str or not merge_date_str:
    return
```